beacon-chain/powchain/service.go

```diff
@@ -153,6 +153,43 @@ type Web3ServiceConfig struct {
 	Eth1HeaderReqLimit uint64
 }
 
+// HttpEndpoint is an endpoint with authorization data.
+type HttpEndpoint struct {
```

Comment:
Is it possible to have some sort of better abstraction here to differentiate a raw endpoint vs auth endpoint ? If we ever add in other forms of auth there will be a lot of `if else` checks to determine what to do